.github/workflows/build.yml

```diff
@@ -36,6 +36,6 @@ jobs:
         uses: docker/build-push-action@v5
         with:
           push: true
-          tags: ghcr.io/${{ github.repository }}/template-component-package:latest
+          tags: ghcr.io/${{ github.repository_owner }}/template-component-package:latest
```

Comment:
We still have to agree on something for here.

Reply:
To be honest, maybe we need to be more clever here anyway because already when the repo is created, this will build and push an image where the user didn't even have the chance to rename yet

Reply:
> To be honest, maybe we need to be more clever here anyway because already when the repo is created, this will build and push an image where the user didn't even have the chance to rename yet

Good point. Maybe we name the file `.github/workflows/build.yml.template` and let the rename script rename it to `yml` once it has replace the name?

Reply:
Or we make it on workflow dispatch... Full control to the user

Reply:
Apart from that there is something else missing no? https://github.com/domire8/intelligent-component/actions/runs/11455380338/job/31871271215

Do we need to login to GHCR?

Reply:
> Or we make it on workflow dispatch... Full control to the user

Also this because currently it's `on: push` so on push on every branch...

Reply:
> Or we make it on workflow dispatch... Full control to the user

They need to write an extra file, I don't mind that but I feel like a template is an easier barrier to entry.

> Also this because currently it's `on: push` so on push on every branch...

We can restrict it to `main`, this is pretty standard on GitHub (and most of their default workflows do that).

```yaml
on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]
```

> Apart from that there is something else missing no? https://github.com/domire8/intelligent-component/actions/runs/11455380338/job/31871271215

The URL must be in the format `ghcr.io/ORG/PACKAGE_NAME:TAG` you have `ghcr.io/ORG/REPO/PACKAGE_NAME:TAG` which is invalid.